sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -498,6 +500,19 @@ def test_sparse_coder_estimator():
     assert np.sqrt(np.sum((np.dot(code, V) - X) ** 2)) < 0.1
 
 
+def test_sparse_coder_estimator_clone():
+    n_components = 12
+    rng = np.random.RandomState(0)
+    V = rng.randn(n_components, n_features)  # random init
+    V /= np.sum(V ** 2, axis=1)[:, np.newaxis]
+    code = SparseCoder(dictionary=V, transform_algorithm='lasso_lars',
+                       transform_alpha=0.001)
+    cloned = clone(code)
+    assert cloned.dictionary.tolist() == code.dictionary.tolist()
```

Comment:
Since we are cloning, I assume that we want to check that we don't have the same `id`?

Reply:
Fixed.

Reply:
I assume @glemaitre meant:

```
assert id(cloned.dictionary) != id(code.dictionary)
```

which should be the case when using `clone(est, safe=True)` which is the default. For instance it is the case for KMeans:


```python
>>> import numpy as np                                                                           
>>> from sklearn.base import clone                                                               
>>> from sklearn.cluster import KMeans                                                           
>>> init = np.random.randn(100, 10)                                                              
>>> km = KMeans(n_clusters=init.shape[0], init=init)                                             
>>> cloned_km = clone(km)                                                                        
>>> id(cloned_km.init) != id(km.init)                                                            
True
```